### PR TITLE
[IMPL] Texture3D Sample

### PR DESCRIPTION
**Pull request by IngmarBitter**
[IMPL] Texture3d Sample

A sample that uses a 256x256x256 3d texture that is the RGB color cube and places a simple shape to cut through it.

### CODE REVIEW

File: samples/TerraFX/Graphics/HelloTexture3d.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        }
+
+        public override void Initialize(Application application)
+        {
+            base.Initialize(application);
+
+            var graphicsDevice = GraphicsDevice;
+
+            using (var vertexStagingBuffer = graphicsDevice.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Default, GraphicsResourceCpuAccess.Write, 64 * 1024))
+            using (var textureStagingBuffer = graphicsDevice.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Default, GraphicsResourceCpuAccess.Write, 64 * 1024 * 4))
```
Comment (tannergooding) [inline samples/TerraFX/Graphics/HelloTexture3d.cs:33]: As mentioned on Discord, this should be `64 * 1024 * 1024` so the buffer is large enough to hold the entire 3D texture.

Comment (tannergooding) [reply samples/TerraFX/Graphics/HelloTexture3d.cs:33]: Once that is resolved, there is a separate issue where `D3D12GraphicsTexture` always sets `shaderResourceViewDesc` to `D3D12_SRV_DIMENSION_TEXTURE2D`.

I'll push a fix for this tomorrow.

Comment (IngmarBitter) [reply samples/TerraFX/Graphics/HelloTexture3d.cs]: ok

**Comment by IngmarBitter** [comment]
my last push was just to rebase onto the current upstream/main

**Review by tannergooding — APPROVED** [review]

**Pull request merged**